common/diagnostics/Diagnostics.java

```diff
@@ -128,16 +144,25 @@ protected static void initSentry(String serverID, String distributionName, Strin
         }
 
         @Override
-        public void mayStartServing(@Nullable SslContext sslContext, ChannelInboundHandlerAdapter... middleware) {
+        public void mayStartMonitoring(@Nullable SslContext sslContext, ChannelInboundHandlerAdapter... middleware) {
             if (monitoringServer != null) monitoringServer.startServing(sslContext, middleware);
         }
 
         @Override
-        public void submitError(Throwable error) {
+        public void mayStartReporting() {
+            if (statisticReporter == null) {
+                LOG.error("Cannot start reporting because statistic reporter is not initialised yet.");
+                return;
+            }
```

Comment:
A `Diagnostics.Core` should never have a `null` `statisticReporter` in the current state.